presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -107,7 +107,7 @@ public SqlTaskManager(
 
         DataSize maxBufferSize = config.getSinkMaxBufferSize();
 
-        taskNotificationExecutor = newCachedThreadPool(threadsNamed("task-notification-%s"));
+        taskNotificationExecutor = newFixedThreadPool(config.getTaskNotificationThreads(), threadsNamed("task-notification-%s"));
```

Comment:
We should create a bounded thread pool instead so that the threads retire when there is no load.